### PR TITLE
[docs] Prepare docs

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Closes https://github.com/elastic/observability-docs/issues/4056
Closes https://github.com/elastic/obs-docs-team/issues/240

Builds out minimal viable docs.

## What's included

This includes the overall structure of the initial doc set. There is some missing information. I've left code comments with questions to be answered throughout the files.

Read more about what needs to be included in these docs in [Elastic Distributions for OpenTelemetry (Mini docs style guide)](https://docs.google.com/document/d/1W0pvuEdVQpa8_dpiBBO0mQwZXZwTgBu8S4SkV40kjJQ/edit#heading=h.bg370r7gv0lr) (internal doc).

## To do

- [x] @colleenmcginnis convert back to AsciiDoc
- [x] @colleenmcginnis add initial outline with questions for the dev team in code comments
- [x] @colleenmcginnis continue clean up / aligning across languages
- [x]  @elastic/apm-agent-net review content that's already there, add content that's missing (especially where there are questions in code comments)
  * Feel free to either commit directly to this PR or add suggestions in PR comments &mdash; whichever works best for you! 
- [x] @colleenmcginnis clean up content
- [x] @elastic/obs-docs peer review
- [x] @colleenmcginnis address feedback
- [x] 🆕  @colleenmcginnis convert to Markdown
- [x] 🆕  @colleenmcginnis reorganize content as needed to be consumed on GitHub
- [x] 🆕  @colleenmcginnis slight content reframing
- [x]  🆕 @colleenmcginnis update product naming as needed
- [x]  @elastic/apm-agent-net final review, approve
- [x] @colleenmcginnis merge

### CODE REVIEW

**Comment by colleenmcginnis** [comment]
👋 @elastic/apm-agent-net can you please take an initial look here? I left _many_ questions for you in code comments throughout the doc files. As you read through, if you have answers to questions, feel free to either commit directly to this PR or add suggestions in PR comments — whatever works best for you!

File: docs/configure.asciidoc [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+:language: .NET
+:language_lc: dotnet
+:distro_name: Elastic Distribution for OpenTelemetry {language}
+
+include::release-status.asciidoc[]
+
+The {distro_name} ("the distro") offers a lot of flexibility...
+
+////
+How do users set configuration options?
```
Comment (Mpdreamz) [inline docs/configure.asciidoc:22]: Either through environment variables or `IConfiguration` the latter is a well known configuration method for .NET apps.

The source for which can be files/databases etc whatever the user choses. The example in the document uses a settings json file which is the most common.

File: docs/configure.asciidoc [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+////
+What are the available configuration options?
+////
+[discrete]
+[[configure-configuration-options]]
+=== Configuration options
+
+////
+Is this true? Is the distro a wrapper around the OpenTelemetry .NET agent
```
Comment (Mpdreamz) [inline docs/configure.asciidoc:160]: The OpenTelemetry .NET Client is an implementation of the general OpenTelemetry SDK guidelines not a wrapper. 

I think its best to say we are an open extension of the OpenTelemetry .NET SDK and are therefor not limiting anything from the standard SDK.

So we are an extension from an implementation. Not a wrapper of a wrapper, this indeed sound like we are hiding/obscuring this which we are **actively** not 😸 

https://opentelemetry.io/docs/specs/otel/overview/

File: docs/configure.asciidoc [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+* General OpenTelemetry SDK configuration options
+* Elastic-specific configuration options that are only available when using the distro
+
+[discrete]
+[[configure-otel-sdk-options]]
+==== OpenTelemetry SDK configuration options
+
+////
+Is this true? Are there any options that aren't supported?
```
Comment (Mpdreamz) [inline docs/configure.asciidoc:173]: Yes

File: docs/configure.asciidoc [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+Any options that shouldn't be used?
```
Comment (Mpdreamz) [inline docs/configure.asciidoc:174]: No, we actively don't want to limit this. As of today we are not aware of any option that should not be used.

File: docs/get-started.asciidoc [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+:language: .NET
+:language_lc: dotnet
+:distro_name: Elastic Distribution for OpenTelemetry {language}
+
+include::release-status.asciidoc[]
+
+This guide shows you how to use the {distro_name} ("the distro") to instrument your {language} application and send OpenTelemetry data to an Elastic Observability deployment.
+
+////
+Is this true?
```
Comment (Mpdreamz) [inline docs/get-started.asciidoc:23]: Is what true? Sorry not sure what this question pertains

File: docs/get-started.asciidoc [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+host-based applications like https://learn.microsoft.com/en-us/dotnet/core/extensions/workers[worker services].
+
+[NOTE]
+====
+The examples below assume your code uses the top-level statements feature introduced in C# 9.0
+and the default choice for applications created using the latest templates.
+====
+
+////
+Are there any dependencies that need to be installed in addition to the distro?
```
Comment (Mpdreamz) [inline docs/get-started.asciidoc:126]: No and no order does not matter

File: docs/get-started.asciidoc [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+});
+----
+<1> Using this URL will require two redirects, allowing us to see multiple spans in the trace.
+
+The distro will automatically enable the exporting of signals via the OTLP exporter. This
+exporter requires that endpoint(s) are configured. A common mechanism for configuring
+endpoints is via environment variables.
+
+////
+What does the _minimal_ configuration / setup look like?
```
Comment (Mpdreamz) [inline docs/get-started.asciidoc:213]: The next section mentions the minimum required config

File: docs/get-started.asciidoc [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+
+. Configure environment variables for the application either in `launchSettings.json` or in the environment
+where the application is running.
+. Once configured, run the application and make a request to the root endpoint. A trace will be generated
+and exported to the OTLP endpoint.
+
+For more information on all the available configuration options, refer to <<configure>>.
+
+////
+Does the user need to do something after configuring the distro
```
Comment (Mpdreamz) [inline docs/get-started.asciidoc:254]: no

File: docs/get-started.asciidoc [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+[discrete]
+[[get-started-initialization]]
+==== Initialize the distro
+
+////
+Step-by-step instructions
+////
+
+////
+What should they expect to see after running this?
```
Comment (Mpdreamz) [inline docs/get-started.asciidoc:267]: Not sure I understand, the next section has a successful trace in the waterfall view in the Elastic APM UI. This should suffice?

**Comment by Mpdreamz** [comment]
Hey @colleenmcginnis thanks for getting our docs in shape. 

I addressed several questions but its hard to parse whats a question to us vs whats a structural comment that describes the paragraph.

Let me know if I missed any! 

It would be good to get this in soonish so that we can do another pass over the content as part of #126.

**Comment by colleenmcginnis** [comment]
Thanks @Mpdreamz. 

>I addressed several questions but its hard to parse whats a question to us vs whats a structural comment that describes the paragraph.

Sorry for the confusion!

File: docs/configure.asciidoc (hunk 2 above)
Comment (colleenmcginnis) [reply docs/configure.asciidoc:160]: This is a very helpful clarification. Thanks!

File: docs/get-started.asciidoc (hunk 5 above)
Comment (colleenmcginnis) [reply docs/get-started.asciidoc:23]: Is this true and it is something we want to include in the docs?

>As an OpenTelemetry SDK, the distro supports sending data to any OpenTelemetry protocol (OTLP) endpoint ([OpenTelemetry Collector](https://opentelemetry.io/docs/collector/)) [but this guide assumes you are sending data to Elastic].

(Context: This came up while I was standardizing the docs across languages. It was mentioned in the Node.js docs.  I'm wondering if this is true of all the distros _and_ if it's worth mentioning.

Comment (Mpdreamz) [reply docs/get-started.asciidoc:23]: This is true and love the addition 👍

File: docs/configure.asciidoc [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+Environment variables are read at startup and can be used to configure the Elastic distribution.
+For details of the various options available and their corresponding environment variable names,
+see <<configure-configuration-options>>.
+
+// Order of operations
+//
+// TO DO: What if a configuration option is set manually in the code?
+// Would that take precedence over environment variables and/or the `IConfiguration` system?
+Environment variables always take precedence over configuration provided by the `IConfiguration`
+system.
```
Comment (colleenmcginnis) [inline docs/configure.asciidoc:56]: In the docs we mention:

>Environment variables always take precedence over configuration provided by the `IConfiguration`
system.

What if a configuration option is set manually in the code? Would that value take precedence over environment variables and/or the `IConfiguration` system? Or the opposite?

**Review by colleenmcginnis — COMMENTED** [review]
@Mpdreamz one more question for now!

File: docs/configure.asciidoc (hunk 10 above)
Comment (Mpdreamz) [reply docs/configure.asciidoc:56]: Yep!

Code Config > Environment config > IConfiguration

File: docs/get-started.asciidoc [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+✅ What the user needs to know and/or do before they install the distro
+////
+[discrete]
+[[get-started-prerequisites]]
+=== Prerequisites
+
+Before getting started:
+
+// Familiarity with OpenTelemetry
+* *Familiarize yourself with the OpenTelemetry SDK.* The distro is a lightweight wrapper around the OpenTelemetry SDK. You should be broadly familiar with the OpenTelemetry SDK concepts and instrumenting applications using the Microsoft diagnostic APIs. If you are not familiar with this, start with the https://opentelemetry.io/docs/languages/net[OpenTelemetry SDK documentation].
```
Comment (colleenmcginnis) [inline docs/get-started.asciidoc:37]: This content was in the getting started guide before I started making edits. I carried over the "Familiarize yourself with the OpenTelemetry SDK" prerequisite to the Java and Node.js getting started guides, too. However, @trentm brought up an interesting point in the Node.js PR:

>I'm concerned that saying a new user needs to familiarize with the OTel SDK before using our distro will somewhat scare them off. The ideal golden path is for users to just install our distro package, set 2 or 3 environment variables to configure it to run with their app, and then see tracing/metrics/logs in their Elastic instance -- no real familiarity with OpenTelemetry required.

For the .NET distro, do users really need to be broadly familiar with the OpenTelemetry SDK, or can they build up that knowledge later if/when they want to customize how they're using the distro?

Comment (Mpdreamz) [reply docs/get-started.asciidoc:37]: Its a tight rope to walk indeed!

Our distribution are as plug and play as possible yet still exposing the full power of the underlying OpenTelemetry infrastructure. That's part of our value add.

I would refrain from calling us a `wrapper` since it might indicate us masking features. 

Something like:

> The Elastic Distribution is an open extension of the OpenTelemetry SDK. It's goal is to ship an easy to install Observability setup with the right defaults without restricting access to the underlying OpenTelemetry SDK concepts. 
We encourage you to read the https://opentelemetry.io/docs/languages/net[OpenTelemetry SDK documentation] if you need tighter control over your configuration.  

perhaps?

File: docs/get-started.asciidoc [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+[NOTE]
+.The Elastic distro and the OpenTelemetry SDK
+====
+The distro includes a transitive dependency on the OpenTelemetry SDK, so you do not _need_ to add the
+OpenTelemetry SDK package to your project directly. However, you _can_ explicitly add the OpenTelemetry SDK as a dependency if you want to opt into newer SDK versions. If you do this, the OpenTelemetry SDK dependency must
+be defined _before_ the Elastic OpenTelemetry Distribution for .NET is defined.
+====
+
+[NOTE]
+.Migrating from the OpenTelemetry SDK to the Elastic distro
```
Comment (bmorelli25) [inline docs/get-started.asciidoc:87]: I'm curious about this. On one hand, I see why it fits here. On the other, I can see it being its own section or own page. I guess it's a really small update if someone is migrating which is why you've added it in the way you did.

Comment (bmorelli25) [reply docs/get-started.asciidoc:87]: Not sure what I'm trying to say. Likely fine as-is here.

Comment (colleenmcginnis) [reply docs/get-started.asciidoc:87]: Yea. The Java docs has a full page for migrating from the APM agent to the Elastic distro, these docs have this note about going from the Otel SDK to the Elastic distro, and the Node docs have neither. At least for 8.15, I've been [considering migration content as optional](https://docs.google.com/document/d/1W0pvuEdVQpa8_dpiBBO0mQwZXZwTgBu8S4SkV40kjJQ/edit#heading=h.afljcowgd5on) (i.e. not a part of "minimal viable docs"), but I'm open to other opinions.

Comment (bmorelli25) [reply docs/get-started.asciidoc:87]: No, I think that's the right call. If the experience is different then there's no need for us to standardize the approach we take to documenting it.

File: docs/get-started.asciidoc [hunk 13]
```diff
@@ -0,0 +1,265 @@
...
+This package includes instrumentation to collect traces for requests handled by ASP.NET Core endpoints.
++
+[NOTE]
+====
+The ASP.NET Core instrumentation is not included by default in the distro.
+As with all optional instrumentation libraries, you can choose to include them in your application by
+adding a suitable package reference.
+====
+
+. Inside the `Program.cs` file of the ASP.NET Core application, add the following two using directives:
```
[the remark this answers is not in the document]
Comment (colleenmcginnis) [reply docs/get-started.asciidoc:143]: I don't _think_ so, but maybe formatting `using` ([.NET docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive)) would help make it clearer:

>1. Inside the `Program.cs` file of the ASP.NET Core application, add the following two `using` directives:
>
>```csharp
>using OpenTelemetry;
>using OpenTelemetry.Trace;
>```

[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/get-started.asciidoc:143]: OH! I get it now. Yes, I think so.

**Comment by Mpdreamz** [comment]
@colleenmcginnis let me know when you need us to do the final review for that 2nd last checkmark 😸

**Comment by colleenmcginnis** [comment]
Thanks for your patience here! I've updated the "To do" list above with some 🆕  items that reflect some recent changes to how we're approaching documenting Elastic Distributions for OpenTelemetry. This includes converting from AsciiDoc to Markdown. Users will be directed to this repo to read the docs on GitHub. We're _not_ going to build and host these docs on elastic.co/guide right now.

**Comment by colleenmcginnis** [comment]
@Mpdreamz  I split the [naming changes](https://github.com/elastic/apm-dev/issues/1080#issuecomment-2286361572) into two commits: [`update naming (docs files)`](https://github.com/elastic/elastic-otel-dotnet/pull/127/commits/1de0ef3ca2850a58bd54c4cbc04d5b194a0778fc) and [`update naming (non-docs files)`](https://github.com/elastic/elastic-otel-dotnet/pull/127/commits/d023effad0a9707f3f72dba8fab92e5b945aa516) so we can easily revert if you want to handle non-docs files separately.

File: docs/get-started.md [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+> EDOT .NET is designed to be easy to use and integrate into your applications. This includes applications that have previously used the OpenTelemetry SDK directly. In situations where the OpenTelemetry SDK is already used, the only required change is to add the `Elastic.OpenTelemetry` NuGet package to the project. Doing so will automatically switch to the opinionated configuration provided by EDOT .NET.
+
+<!-- ✅ Start-to-finish operation -->
+## Instrument your application
+
+The OpenTelemetry SDK and EDOT .NET provide extension methods to enable observability features in your application with a few lines of code. Choose from:
+
+* [**ASP.NET Core application**](#aspnet-core-application) for ASP.NET Core minimal API applications, other ASP.NET Core workloads and other host-based applications like [worker services](https://learn.microsoft.com/en-us/dotnet/core/extensions/workers).
+* [**`Microsoft.Extensions.Hosting`**](#microsoftextensionshosting) for console applications and services that are written against a builder that exposes an `IServiceCollection`.
+* [**Manual instrumentation**](#manual-instrumentation) for environments where an `IServiceCollection` is unavailable.
```
Comment (colleenmcginnis) [inline docs/get-started.md:73]: Does this framing make sense? Since we're keeping the content in GitHub (not publishing to the Elastic docs site), I attempted to combine the existing README content with the content in the docs directory.

File: docs/configure.md [hunk 15]
```diff
@@ -0,0 +1,262 @@
...
+
+// Create an instance of `ElasticOpenTelemetryBuilderOptions`.
+var builderOptions = new ElasticOpenTelemetryBuilderOptions
+{
+  // Create an instance of `ElasticOpenTelemetryOptions` and configure
+  // the file log directory by setting the corresponding property.
+	DistroOptions = new ElasticOpenTelemetryOptions
+	{
+    // This example sets the file log directory to `C:\Logs`
+    // which enables diagnostic file logging.
```
Comment (Mpdreamz) [inline docs/configure.md:114]:
```suggestion
	// This example sets the file log directory to `C:\Logs`
	// which enables diagnostic file logging.
```

File: docs/configure.md [hunk 16]
```diff
@@ -0,0 +1,262 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+
+var services = new ServiceCollection();
+
+// Create an instance of `ElasticOpenTelemetryBuilderOptions`.
+var builderOptions = new ElasticOpenTelemetryBuilderOptions
+{
+  // Create an instance of `ElasticOpenTelemetryOptions` and configure
+  // the file log directory by setting the corresponding property.
```
Comment (Mpdreamz) [inline docs/configure.md:110]:
```suggestion
	// Create an instance of `ElasticOpenTelemetryOptions` and configure
	// the file log directory by setting the corresponding property.
```

File: docs/get-started.md (hunk 14 above)
Comment (Mpdreamz) [reply docs/get-started.md:73]: This works for now I'll reword these bullet points in a follow up PR (and ask for your review instead :)).

I need to add more content around `Automatic Instrumentation` that recently landed as a feature.

File: docs/get-started.md [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+    This package includes instrumentation to collect traces for requests handled by ASP.NET Core endpoints.
+
+    > [!NOTE]
+    > The ASP.NET Core instrumentation is not included by default in EDOT .NET. As with all optional instrumentation libraries, you can choose to include them in your application by adding a suitable package reference.
+
+1. Inside the `Program.cs` file of the ASP.NET Core application, add the following two `using` directives:
+
+    ```csharp
+    using OpenTelemetry;
+    using OpenTelemetry.Trace;
```
Comment (Mpdreamz) [inline docs/get-started.md:99]:
```suggestion
using OpenTelemetry;
using OpenTelemetry.Trace;
```

File: docs/get-started.md [hunk 18]
```diff
@@ -0,0 +1,237 @@
...
+      // The `AddHttpClient` method registers the `IHttpClientFactory` service with
+      // the dependency injection container. This is NOT required to enable OpenTelemetry,
+      // but the example endpoint will use it to send an HTTP request.
+      .AddHttpClient()
+      // The `AddOpenTelemetry` method registers the OpenTelemetry SDK with the
+      // dependency injection container. When available, EDOT .NET will override
+      // this to add opinionated defaults.
+      .AddOpenTelemetry()
+        // Configure tracing to instrument requests handled by ASP.NET Core.
+        .WithTracing(t => t.AddAspNetCoreInstrumentation());
```
Comment (Mpdreamz) [inline docs/get-started.md:119]:
```suggestion
var builder = WebApplication.CreateBuilder(args);

builder.Services
	// The `AddHttpClient` method registers the `IHttpClientFactory` service with
	// the dependency injection container. This is NOT required to enable OpenTelemetry,
	// but the example endpoint will use it to send an HTTP request.
	.AddHttpClient()
	// The `AddOpenTelemetry` method registers the OpenTelemetry SDK with the
	// dependency injection container. When available, EDOT .NET will override
	// this to add opinionated defaults.
	.AddOpenTelemetry()
	// Configure tracing to instrument requests handled by ASP.NET Core.
	.WithTracing(t => t.AddAspNetCoreInstrumentation());
```

File: docs/get-started.md [hunk 19]
```diff
@@ -0,0 +1,237 @@
...
+To demonstrate the tracing capabilities, add a simple endpoint to the application:
+
+```csharp
+app.MapGet("/", async (IHttpClientFactory httpClientFactory) =>
+{
+	using var client = httpClientFactory.CreateClient();
+
+	await Task.Delay(100);
+  // Using this URL will require two redirects, allowing us to
+  // see multiple spans in the trace.
```
Comment (Mpdreamz) [inline docs/get-started.md:135]:
```suggestion
	// Using this URL will require two redirects, allowing us to
	// see multiple spans in the trace.
```

File: docs/get-started.md [hunk 20]
```diff
@@ -0,0 +1,237 @@
...
+```
+
+### Manual instrumentation
+
+In environments where an `IServiceCollection` is unavailable you may manually start instrumenting by creating an instance of `ElasticOpenTelemetryBuilder`.
+
+```csharp
+await using var session = new ElasticOpenTelemetryBuilder()
+    .WithTracing(b => b.AddSource(ActivitySourceName))
+    .Build();
```
Comment (Mpdreamz) [inline docs/get-started.md:177]:
```suggestion
	.WithTracing(b => b.AddSource(ActivitySourceName))
	.Build();
```

File: docs/get-started.md [hunk 21]
```diff
@@ -0,0 +1,237 @@
...
+1. Scroll down and select the **OpenTelemetry** option.
+1. The appropriate values for `OTEL_EXPORTER_OTLP_ENDPOINT` and `OTEL_EXPORTER_OTLP_HEADERS` are shown there.
+
+    ![Elastic Cloud OpenTelemetry configuration](images/elastic-cloud-opentelemetry-configuration.png)
+
+    For example:
+
+    ```sh
+    export OTEL_EXPORTER_OTLP_ENDPOINT=https://my-deployment.apm.us-west1.gcp.cloud.es.io
+    export OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer P....l"
```
Comment (Mpdreamz) [inline docs/get-started.md:215]:
```suggestion
export OTEL_EXPORTER_OTLP_ENDPOINT=https://my-deployment.apm.us-west1.gcp.cloud.es.io
export OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer P....l"
```

**Review by Mpdreamz — CHANGES_REQUESTED** [review]
@colleenmcginnis almost LGTM 😸 added a few spacing nitpicks.

**Comment by colleenmcginnis** [comment]
>added a few spacing nitpicks.

Oh gosh this must have been my local whitespace settings. 🙈

File: docs/get-started.md (hunk 14 above)
Comment (colleenmcginnis) [reply docs/get-started.md:73]: Sounds good!

**Review by Mpdreamz — APPROVED** [review]
